web/src/features/panels/asset/SolarAssetPanel.tsx

```diff
@@ -0,0 +1,207 @@
+import { useAtomValue, useSetAtom } from 'jotai';
+import { ArrowLeft } from 'lucide-react';
+import { useLocation } from 'react-router-dom';
+import { useNavigateWithParameters } from 'utils/helpers';
+
+import { getStatusColor } from '../../assets/utils';
+import { selectedSolarAssetAtom } from '../../map/mapAtoms';
+import GenericPanel from '../InterfacePanel';
+
+// Helper to format date string (YYYY-MM-DDTHH:mm:ssZ) to MonthName Day, Year
+const formatDate = (dateString: string | undefined | null) => {
+  if (!dateString) {
+    return 'N/A';
+  }
+  try {
+    const date = new Date(dateString);
+    return date.toLocaleDateString('en-US', {
+      month: 'long',
+      day: 'numeric',
+      year: 'numeric',
+    });
+  } catch {
+    return 'Invalid Date';
+  }
+};
+
+export default function SolarAssetPanel() {
+  const selectedAsset = useAtomValue(selectedSolarAssetAtom);
+  const setSelectedAsset = useSetAtom(selectedSolarAssetAtom);
+  const navigate = useNavigateWithParameters();
+  const location = useLocation();
+
+  if (!selectedAsset) {
+    return null;
+  }
+
+  console.log('random');
+  const { properties } = selectedAsset;
+
+  // Debug all properties to find potential URL fields
+  console.log('[SolarAssetPanel] All properties:', properties);
+  const possibleUrlFields = Object.entries(properties).filter(
+    ([_, value]) =>
+      typeof value === 'string' &&
+      (value.startsWith('http://') ||
+        value.startsWith('https://') ||
+        value.startsWith('www.'))
+  );
+  console.log('[SolarAssetPanel] Possible URL fields:', possibleUrlFields);
+
+  const handleClose = () => {
```

Comment:
This needs to be wrapped in a useCallback since it's later passed to a component and declared inside a component. Otherwise it will cause performance issues as it gets recreated on every render which cascades down if the reference is not stable.